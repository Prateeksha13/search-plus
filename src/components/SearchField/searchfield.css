.search-field-wrapper {
  display: flex;
  position: relative;
}

.search-field-icon {
  position: absolute;
  top: 14px;
  left: 16px;
  z-index: 1;
}

.react-autosuggest__container {
  position: relative;
  width: 100%;
}

.react-autosuggest__input {
  width: 100%;
  padding: 12px 16px 12px 44px;
  border-radius: 12px;
  background: var(--white);
  border: 1px solid var(--border-dark);
  font-weight: normal;
  font-size: 17px;
  text-align: left;
  color: var(--text-color);
  box-sizing: border-box;
  -webkit-appearance: none;
}

.react-autosuggest__input--focused {
  outline: none;
  border-color: var(--border-blue);
}

.react-autosuggest__input--open {
  border-radius: 15px 15px 0 0;
}

.react-autosuggest__input::-ms-clear {
  display: none;
}

.react-autosuggest__suggestions-container {
  display: none;
}

.react-autosuggest__suggestions-container--open {
  display: block;
  position: absolute;
  top: 44px;
  width: 100%;
  border: 1px solid var(--border-dark);
  background-color: var(--white);
  font-size: 17px;
  z-index: 2;
  border-radius: 0 0 12px 12px;
  overflow: hidden;
  width: calc(100% - 2px);
}

.react-autosuggest__suggestions-list {
  margin: 0;
  padding: 0;
  list-style-type: none;
}

.react-autosuggest__suggestion {
  cursor: pointer;
  padding: 16px 16px;
}

.react-autosuggest__suggestion--highlighted {
  background-color: #ddd;
}

.react-autosuggest__section-container {
  border-top: 1px dashed #ccc;
}

.react-autosuggest__section-container--first {
  border-top: 0;
}

.react-autosuggest__section-title {
  padding: 10px 0 0 10px;
  font-size: 12px;
  color: #777;
}

.autocomplete-wrapper .autocomplete-label {
  font-size: 0.8rem;
  font-weight: bold;
  color: #07032b;
  color: var(--text-color);
  margin-bottom: 16px;
  display: block;
  letter-spacing: 0.02rem;
}
