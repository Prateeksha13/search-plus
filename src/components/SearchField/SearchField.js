import React, { useState } from "react";
import Autosuggest from "react-autosuggest";

import SearchIcon from "../../assets/forms/search.svg";
import "./searchfield.css";

const SearchField = ({
  informSuggestionSelect,
  getSuggestions,
  renderSuggestion,
  getSuggestionValue,
  label,
  placeholder,
  inputRegex,
  shouldAutoFocus,
  onInputChange,
}) => {
  const [value, setValue] = useState("");
  const [suggestions, setSuggestions] = useState([]);

  // When suggestion is clicked, Autosuggest needs to populate the input
  // based on the clicked suggestion. Teach Autosuggest how to calculate the
  // input value for every given suggestion.

  const onChange = (event, { newValue }) => {
    if (inputRegex) {
      if (inputRegex.test(newValue)) {
        setValue(newValue);
      }
    } else {
      setValue(newValue);
    }
    onInputChange(newValue);
  };

  // Autosuggest will call this function every time you need to update suggestions.
  const onSuggestionsFetchRequested = ({ value }) => {
    setValue(value);
    setSuggestions(getSuggestions(value));
  };

  // Autosuggest will call this function every time you need to clear suggestions.
  const onSuggestionsClearRequested = () => {
    setSuggestions([]);
  };

  const onSuggestionSelected = (
    event,
    { suggestion, suggestionValue, suggestionIndex, sectionIndex, method }
  ) => {
    informSuggestionSelect(suggestion, suggestionValue);
  };

  return (
    <div className="search-field-wrapper">
      {label && <label className="search-field-label">{label}</label>}
      <img className="search-field-icon" src={SearchIcon} alt="search" />
      <Autosuggest
        highlightFirstSuggestion={true}
        suggestions={suggestions}
        onSuggestionsFetchRequested={onSuggestionsFetchRequested}
        onSuggestionsClearRequested={onSuggestionsClearRequested}
        getSuggestionValue={getSuggestionValue}
        renderSuggestion={renderSuggestion}
        inputProps={{
          placeholder: placeholder ? placeholder : "Search",
          value: value,
          onChange: onChange,
          autoFocus: shouldAutoFocus,
          // onBlur: (event) => informSuggestionSelect(event.target.value),
        }}
        onSuggestionSelected={onSuggestionSelected}
      />
    </div>
  );
};

export default SearchField;
