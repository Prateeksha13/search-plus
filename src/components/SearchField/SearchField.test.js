import renderer from "react-test-renderer";
import SearchField from "./SearchField";

it("changes the class when hovered", () => {
  const component = renderer.create(
    <SearchField
      informSuggestionSelect={() => {
        console.log("");
      }}
      getSuggestions={() => {
        console.log("");
      }}
      renderSuggestion={() => {
        console.log("");
      }}
      getSuggestionValue={() => {
        console.log("");
      }}
      label="Search"
      placeholder="Search by product"
      inputRegex=""
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  // renderer.act(() => {
  //   tree.props.onMouseEnter();
  // });
  // re-rendering
  // tree = component.toJSON();
  // expect(tree).toMatchSnapshot();

  // // manually trigger the callback
  // renderer.act(() => {
  //   tree.props.onMouseLeave();
  // });
  // // re-rendering
  // tree = component.toJSON();
  // expect(tree).toMatchSnapshot();
});
