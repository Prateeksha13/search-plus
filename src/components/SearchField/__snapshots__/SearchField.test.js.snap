// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the class when hovered 1`] = `
<div
  className="search-field-wrapper"
>
  <label
    className="search-field-label"
  >
    Search
  </label>
  <img
    alt="search"
    className="search-field-icon"
    src="search.svg"
  />
  <div
    aria-expanded={false}
    aria-haspopup="listbox"
    aria-owns="react-autowhatever-1"
    className="react-autosuggest__container"
    role="combobox"
  >
    <input
      aria-activedescendant={null}
      aria-autocomplete="list"
      aria-controls="react-autowhatever-1"
      autoComplete="off"
      className="react-autosuggest__input"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      placeholder="Search by product"
      type="text"
      value=""
    />
    <div
      className="react-autosuggest__suggestions-container"
      id="react-autowhatever-1"
      role="listbox"
    />
  </div>
</div>
`;
